browser/js/calculator.js

```diff
@@ -0,0 +1,265 @@
+        this.shrinkFontSize()
+      }
+      this.dom.querySelector( Calculator.elementSelectors.OUTPUT ).innerText = text
+    } else {
+      if ( this.data.current.length > this.data.MAX_LENGTH ) {
+        this.shrinkFontSize()
+        this.shrinkFontSize()
+      }
+      this.dom.querySelector( Calculator.elementSelectors.OUTPUT ).innerText = this.data.current
+    }
+  }
+  Calculator.prototype.setResult = function ( number ) {
+    this.data.result = number.toString()
+    this.data.memory = this.data.result
+  }
+  Calculator.prototype.clearResult = function () {
+    this.data.result = ''
+  }
+  Calculator.prototype.resetData = function () {
+    this.data = {
+      memory: '0',
+      current: '0',
+      operation: '',
+      result: '',
+      MAX_LENGTH: 11
+    }
+  }
+  Calculator.prototype.shrinkFontSize = function () {
+    this.dom.querySelector( Calculator.elementSelectors.OUTPUT )
+      .classList.add( Calculator.cssClasses.SMALL_FONT )
+  }
+  Calculator.prototype.resetFontSize = function () {
+    this.dom.querySelector( Calculator.elementSelectors.OUTPUT )
+      .classList.remove( Calculator.cssClasses.SMALL_FONT )
+  }
+  Calculator.prototype.enterDigit = function ( number ) {
+    var current = this.data.current
+    if ( current === '0' || this.data.memory === this.data.result ) {
+      this.clearResult()
+      this.clearResult()
+      current = ''
+    }
+    if ( this.data.current.length === this.data.MAX_LENGTH ) {
+      this.shrinkFontSize()
+    }
+    current += number
+    this.data.current = current
+    this.setOutput()
+  }
+  Calculator.prototype.setOperator = function ( operation ) {
+    if ( this.data.memory === this.data.result ) {
+      this.data.memory = this.data.result
+    } else {
+      this.data.memory = this.data.current
+    }
+    this.data.current = '0'
+    this.data.operation = operation
+  }
+  Calculator.prototype.clearCalculator = function () {
+    if ( this.data.current === '0' ) {
+      this.resetData()
+      this.setOutput()
+    } else {
+      this.resetFontSize()
+      this.data.current = '0'
+      this.setOutput()
+    }
+  }
+  Calculator.prototype.toggleNegativeValue = function () {
+    if ( this.data.current.indexOf( '-' ) === -1 ) {
+      this.data.current = '-' + this.data.current
+    } else {
+      this.data.current = this.data.current.substring( 1 )
+    }
+    this.setOutput()
+  }
+  Calculator.prototype.convertPercent = function () {
+    if ( this.data.memory === '0' ) {
+      this.data.current = ( parseFloat( this.data.current ) / 100 ).toString()
+      this.setOutput()
+    } else {
+      this.data.current = (
+        (parseFloat( this.data.current ) / parseFloat( this.data.memory )
+      ) * 100 ).toString()
+      this.setOutput()
+    }
+  }
+  Calculator.prototype.addDot = function () {
+    if ( this.data.current.indexOf( '.' ) === -1 ) {
+      this.data.current = this.data.current + '.'
+      this.setOutput()
+    }
+  }
+
```

Comment:
In consistency of spacing in between. It would be great if they were all spaced out.